src/linux.rs

```diff
@@ -1,3 +1,31 @@
+use crate::common;
+
 pub fn run() {
-    println!("Hello, world!");
+    let Ok(bin_path) = std::env::current_exe() else {
+        eprintln!("Failed to get binary path");
+        return;
+    };
+
+    let Some(bin_folder) = bin_path.parent() else {
```

Comment:
These two could be merged into a single call to `std::env::current_dir()`

Reply:
[`current_dir()`](https://doc.rust-lang.org/std/env/fn.current_dir.html) returns the current working directory which is not necessarily the factorio binary folder.

If your current working directory is `~` and you run `./factorio/bin/x64/factorio` the outputs of the functions would be:
| method | value |
| - | - |
| `current_dir()` | `~` |
| `current_exe()` | `~/factorio/bin/x64/factorio` |

_I have not verified if a `LD_PRELOAD` lib on linux would actually return the expected value on linux but I highly assume it does._